### ISSUE TITLE
GR cannot properly save linewidth = 0

### ISSUE THREAD

**Issue by daschw** [JuliaPlots/Plots.jl#1416]
```julia
plot(rand(10), lw = 0)
```
shows as expected in Juno:
<img width="294" alt="gr_linewidth_juno" src="https://user-images.githubusercontent.com/16589944/36779956-ee5e33da-1c71-11e8-96b0-67192f4096a3.PNG">

However when saved as png (or pdf) a thin line appears:
![gr_linewidth](https://user-images.githubusercontent.com/16589944/36780005-16f4d59c-1c72-11e8-9c3c-f5f045789c89.png)

cc: @jheinen

This is particularly annoying for bar plots with a lot of data, e.g.
```julia
bar(rand(10_000), lw = 0, color = :red)
```
which gets saved as
![gr_bar_linewidth](https://user-images.githubusercontent.com/16589944/36780118-5b95b91e-1c72-11e8-9e42-ec634417958e.png)

(A sub-optimal workaround for this last example is to add `lc = :match`)

**Comment by daschw** [comment]
Yes, I would have expected `plot(rand(10), linewidth = 0)` not to show anything, as it is the case in Juno (first image above). Is this not reasonable?
However, if I save this plot as png, I can see a (admittedly very thin) line (second plot above). Would you consider this expected behaviour?

**Comment by jheinen** [comment]
In my underlying GKS, linewidth = 0 means, draw using the thinnest displayable line width. So, for Plots, we have to find another solution.

**Comment by daschw** [comment]
OK, I see, maybe it's just me who would have expected no line for `lw = 0`. In any case, it's not that urgent since `linecolor = invisible()` provides a nice workaround.
